packages/peregrine/lib/talons/CartPage/GiftCards/useGiftCards.js

```diff
@@ -52,10 +52,18 @@ export const useGiftCards = props => {
      *
      * Immediately execute the cart query and set up the other graphql actions.
      */
-    const [getAppliedCards, appliedCardsResult] = useLazyQuery(
-        appliedCardsQuery
-    );
-    const [checkCardBalance, balanceResult] = useLazyQuery(cardBalanceQuery);
+    const appliedCardsResult = useQuery(appliedCardsQuery, {
+        fetchPolicy: 'cache-and-network',
+        nextFetchPolicy: 'cache-first',
+        skip: !cartId,
+        variables: { cartId }
+    });
+
+    const [checkCardBalance, balanceResult] = useLazyQuery(cardBalanceQuery, {
+        // Don't cache this one because the card can be used elsewhere
+        // before it is used again here.
+        fetchPolicy: 'no-cache'
```

Comment:
Nice. Seems like a better place for this. :+1.

Reply:
Yea it didn't make sense to pass as an option to the invoker, and in ACv3 it doesn't seem to work at all - you MUST define this up front.